irods/connection.py

```diff
@@ -325,7 +325,14 @@ def disconnect(self):
                     self.socket = self.socket.unwrap()
                 except AttributeError:
                     pass
-                self.socket.shutdown(socket.SHUT_RDWR)
+                try:
+                    self.socket.shutdown(socket.SHUT_RDWR)
+                except OSError as e:
+                    import errno
+                    # On MacOS this can result in the following error:
+                    # OSError: [Errno 57] Socket is not connected
```

Comment:
This comment should be moved above L329 since that is the function call that triggers this exception block.

Reply:
moved and supplemented.

Reply:
Nice. Resolving.